### PR TITLE
Use light/dark theme based on system preference

### LINKED ISSUE #8183: Enable Auto theming using Dark Mode CSS
For those who don't know,DarkMode CSS is new feature available in all latest browsers(Blink,Gecko,Webkit based).It themes the website based on System's theme.If the system does have Dark theme,then the site would have dark theme,and if the system has light theme ,then the site would have light theme.(One can also create custom behavioyr for each type).
Ref:
1.   [Introduction by Webkit](https://webkit.org/blog/8840/dark-mode-support-in-webkit/)
2.   [Mozilla Docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme)

I want Gitea to have arc-theme(name of dark theme in gitea) when the system is in dark mode and gitea(name of light theme in gitea) when the system is in light mode.

Atleast enable option named "Auto" under themes to enable automatic themes based on dark mode css

### PR DESCRIPTION
**Pull request by gwymor**
Add a new default theme `auto`, which will automatically switch between `gitea` (light) and `arc-green` (dark) themes depending on the user's operating system settings.

Closes: go-gitea#8183

### CODE REVIEW

File: custom/conf/app.example.ini [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ PATH =
 ;DEFAULT_THEME = gitea
 ;;
 ;; All available themes. Allow users select personalized themes regardless of the value of `DEFAULT_THEME`.
-;THEMES = gitea,arc-green
+;THEMES = gitea
```
Comment (delvh) [inline custom/conf/app.example.ini]: I think restricting to one default theme instead of two is a bad idea, especially if there is no way to force one or the other.
I could understand adding this as a third option, but overriding the current behavior in this way is in my opinion not a good idea.
I can (fore)see several problems with this implementation:
1) users can choose their theme themselves. The current implementation will interfere with this mechanism drastically.
2) I think it was explicitly mentioned that `arc-green` should not be seen as **THE** dark theme. It can be treated as a dark theme, but it is specifically not developed as a dark theme. There are other themes for that with more focus on being an actual dark theme.
3) ~~Neither the name `gitea` nor `arc-green` indicate that they are a `light` or a `dark` theme.~~ If I see that correctly, that is exactly the reason why you renamed `arc-green` to `_dark`.
4) I don't think one theme should have two different styles with one style being chosen "randomly". That interferes with my definition of a theme. ("Randomly" as in "you cannot predict which style will be used only by looking at an arbitrary user. You only know that once the client requests the dark theme.")

What I think would be more relevant is to let users choose both a light- and a dark theme themselves and then serve the theme based on the system preference if requested that way, or if configured otherwise the specific theme.
I would say that GitHub shows a good implementation of this, with the only difference being that light- and dark theme can be customized using two of the available themes.
In this implementation, `gitea` can remain the default light theme, and `arc-green` can remain the default dark theme.

**Review by delvh — CHANGES_REQUESTED** [review]

File: custom/conf/app.example.ini (hunk 1 above)
Comment (lafriks) [reply custom/conf/app.example.ini]: I agree there should be additional option `auto`  that can be default

Comment (gwymor) [reply custom/conf/app.example.ini]: I added a new default theme called `auto` which will be the `gitea` theme in light mode and `arc-green` in dark mode.

File: web_src/less/themes/theme-auto.less [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@media (prefers-color-scheme: dark) {
```
Comment (lafriks) [inline web_src/less/themes/theme-auto.less:1]: Is this file needed?

**Comment by delvh** [comment]
Yeah, that implementation is already much better.
However, I still am not a fan of a hard dependency on the `gitea` and the `arc-green` theme as this means that the autodetection mechanism will only be available to these two themes.

As already mentioned, I think the best approach would be to let the user choose two themes instead of one (one as the light theme, the other as the dark theme) in the user settings.
Then, you could add an additional setting to set the preferred theme (`light`, `dark`, or `auto`).

**Comment by lafriks** [comment]
> As already mentioned, I think the best approach would be to let the user choose two themes instead of one (one as the light theme, the other as the dark theme) in the user settings.
> Then, you could add an additional setting to set the preferred theme (`light`, `dark`, or `auto`).

That would have drawback of not able to select other themes (multiple light or dark ones), let's make this simple first as it is now and later on we can improve this

**Comment by delvh** [comment]
What do you mean?
How can you have multiple themes at once?

But in general, I can understand your point.
Okay, let's keep it simple for now.

**Comment by lafriks** [comment]
Not multiple themes at once but allow to server have more than two themes by custom ones and user be able to select any of them

**Comment by delvh** [comment]
I think there is a misunderstanding here.

> As already mentioned, I think the best approach would be to let the user choose two themes instead of one (one as the light theme, the other as the dark theme) in the user settings.
> Then, you could add an additional setting to set the preferred theme (`light`, `dark`, or `system`).

By the second point I mean a mechanism to determine which of the two selected themes to choose:
The dark and light themes are the two previously selected themes. Then, if you choose `system`, depending on your system preference, the light or the dark theme will be displayed.

**Review by axifive — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

File: docs/content/doc/help/faq.en-us.md [hunk 3]
```diff
@@ -158,7 +158,7 @@ Use [Fail2Ban]({{< relref "doc/usage/fail2ban-setup.en-us.md" >}}) to monitor an
 
 ## How to add/use custom themes
 
-Gitea supports two official themes right now, `gitea` and `arc-green` (`light` and `dark` respectively)  
+Gitea supports three official themes right now, `gitea` (light), `arc-green` (dark), and `auto` (automatically switches between the previous two depending on operating system settings).
```
Comment (wxiaoguang) [inline docs/content/doc/help/faq.en-us.md:161]: `auto` is not a real theme, so there are only 2 themes Gitea supports.

**Comment by zeripath** [comment]
make lgtm work

File: web_src/less/themes/theme-auto.less (hunk 2 above)
Comment (clarfonthey) [inline web_src/less/themes/theme-auto.less:1]: Are you sure this is actually right? Doesn't this just not include any CSS for light themes?

Comment (delvh) [reply web_src/less/themes/theme-auto.less:1]: Correct.
That assumes that the light theme only sets color values, and nothing else.
If it sets other values, then we will have hard-to-backtrace bugs.

Comment (clarfonthey) [reply web_src/less/themes/theme-auto.less:1]: Right, what I'm saying is that all CSS is conditional on the dark theme, not just the colours, i.e. it falls back to nothing on light theme. Unless the light theme is somehow also required by the theme.
